Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM sbtscala/scala-sbt:eclipse-temurin-jammy-11.0.17_8_1.9.4_2.13.12
+
+ENV SBT_OPTS="-Xmx4G -XX:+UseG1GC"
+
+WORKDIR /opt/ecos-java-scala
+
+RUN apt-get -yq update && apt-get -yq install cmake build-essential
+
+COPY . .
+
+RUN git submodule update --init --recursive
```

Comment:
@btreecat is docker smart enough to invalidate the change of this layer in case the submodule gets updated? I'm trying to cover edge cases

Reply:
It's not, because we are not versioning the sub-module update. This means docker looks at the command and will just cache it because the command is the same. We can disable cache for the whole build, not ideal but moves it forward.

Reply:
I think I have a solution here, but I am not sure it matters as the ecos repo has not been updated in 9 years.